crates/extensions/tedge_health_ext/src/lib.rs

```diff
@@ -20,26 +21,38 @@ use tedge_mqtt_ext::TopicFilter;
 
 pub struct HealthMonitorBuilder {
     service_name: String,
-    subscriptions: TopicFilter,
     box_builder: SimpleMessageBoxBuilder<MqttMessage, MqttMessage>,
 }
 
 impl HealthMonitorBuilder {
-    pub fn new(service_name: &str) -> Self {
+    pub fn new(
+        service_name: &str,
+        mqtt: &mut (impl ServiceProvider<MqttMessage, MqttMessage, TopicFilter> + AsMut<MqttConfig>),
+    ) -> Self {
+        // Connect this actor to MQTT
         let subscriptions = vec![
             "tedge/health-check",
             &format!("tedge/health-check/{service_name}"),
         ]
         .try_into()
-        .expect("Failed to create the HealthMonitorActor topicfilter");
-        HealthMonitorBuilder {
+        .expect("Failed to create the HealthMonitorActor topic filter");
+
+        let mut box_builder = SimpleMessageBoxBuilder::new(service_name, 16);
+        box_builder
+            .set_request_sender(mqtt.connect_consumer(subscriptions, box_builder.get_sender()));
+
+        let builder = HealthMonitorBuilder {
             service_name: service_name.to_owned(),
-            subscriptions,
-            box_builder: SimpleMessageBoxBuilder::new(service_name, 16),
-        }
+            box_builder,
+        };
+
+        // Update the MQTT config
+        *mqtt.as_mut() = builder.set_init_and_last_will(mqtt.as_mut().clone());
```

Comment:
Although I'm still a bit uncomfortable with this line, and wish that there was a better MQTT config API for an actor to set this config aspect, I'm still glad that this tricky code is now limited to just this function and the usage of the `HealthMonitorBuilder` is minimal and looks very natural from the `main` functions of all the daemons.